lib/PhasarLLVM/TypeHierarchy/DIBasedTypeHierarchy.cpp

```diff
@@ -0,0 +1,272 @@
+      }
+    }
+  }
+
+  // Initialize the transitive closure matrix with all as false
+  llvm::BitVector InitVector(VertexTypes.size(), false);
+
+  for (size_t I = 0; I < VertexTypes.size(); I++) {
+    TransitiveClosure.push_back(InitVector);
+  }
+
+  // set direct edges
+  for (const auto &Vertex : DerivedTypesOf) {
+    TransitiveClosure[Vertex[0]][Vertex[1]] = true;
+  }
+
+  // Add transitive edges
+  bool Change = true;
+  size_t TCSize = TransitiveClosure.size();
+  size_t RowIndex = 0;
+  size_t PreviousRow = 0;
+
+  // (max): I know the code below is very ugly, but I wanted to avoid recursion
+  // if the algorithm goes over the entire matrix and couldn't update any rows
+  // anymore, it stops. As soon as one position gets updated, it goes over the
+  // matrix again
+  while (Change) {
+    Change = false;
+    // go over all rows of the matrix
+    for (size_t CurrentRow = 0; CurrentRow < TCSize; CurrentRow++) {
+      // compare current row with all other rows and check if an edge can be
+      // added
+      for (size_t CompareRow = 0; CompareRow < TCSize; CompareRow++) {
+        // row doesn't need to compare itself with itself
+        if (CurrentRow == CompareRow) {
+          continue;
+        }
+
+        // if the current row is not a parent type of the current compare row,
+        // no edges should be added
+        if (!TransitiveClosure[CurrentRow][CompareRow]) {
+          continue;
+        }
+
+        // Compare both rows and add edges if needed
+        for (size_t Column = 0; Column < TCSize; Column++) {
+          if (TransitiveClosure[CompareRow][Column] &&
+              !TransitiveClosure[CurrentRow][Column] && CurrentRow != Column) {
+            TransitiveClosure[CurrentRow][Column] = true;
+            Change = true;
+          }
+        }
+      }
+    }
+  }
+
+  // add edges onto vertices themselves
+  for (size_t I = 0; I < TransitiveClosure.size(); I++) {
+    TransitiveClosure[I][I] = true;
+  }
+
+  // get VTables
+  const auto Subprograms = Finder.subprograms();
+  for (auto *Subprogram : Subprograms) {
+
+    if (const auto *SubProgramType =
+            llvm::dyn_cast<llvm::DISubprogram>(Subprogram)) {
+      if (!SubProgramType->getVirtuality()) {
+        continue;
+      }
+      // get all virtual functions
+      std::vector<const llvm::Function *> VirtualFunctions;
+      for (const auto &Function : Module->functions()) {
+        if (SubProgramType->getLinkageName() ==
+            Function.stripPointerCasts()->getName().str()) {
+          VirtualFunctions.push_back(&Function);
+        }
+      }
+      VTables.push_back(LLVMVFTable(VirtualFunctions));
+    }
+  }
+}
+
+[[nodiscard]] bool DIBasedTypeHierarchy::isSubType(ClassType Type,
+                                                   ClassType SubType) {
+  // find index of super type
+  unsigned long IndexOfType = TypeToVertex.find(Type)->getSecond();
+  unsigned long IndexOfSubType = TypeToVertex.find(SubType)->getSecond();
```

Comment:
```suggestion
  size_t IndexOfType = TypeToVertex.find(Type)->getSecond();
  size_t IndexOfSubType = TypeToVertex.find(SubType)->getSecond();
```

Please also take care about error handling, i.e. if one of `Type` or `SubType` is not contained in the map